CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.16)
+project(tt-metal)
```

Comment:
Would add some description here following this example. At the very least including VERSION and LANGUAGES
project(tt-metal
  VERSION
    1.0
  DESCRIPTION
    "Tenstorrent Metallium Project"
  LANGUAGES
    CXX
)

https://cmake.org/cmake/help/latest/command/project.html

Reply:
I'm wondering if there's some cmake plugin that we can use to make use of the semantic versioned tags on mainline git branch

Reply:
description added 👍

Reply:
for description, we can do something like this
https://github.com/wolfpld/tracy/blob/master/cmake/version.cmake